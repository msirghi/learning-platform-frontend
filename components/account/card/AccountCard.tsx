import React, { useState } from 'react';
import styles from './AccountCard.module.scss';
import LockIcon from '@material-ui/icons/Lock';
import { IconButton, InputAdornment, Input, Button } from '@material-ui/core';
import EditIcon from '@material-ui/icons/Edit';
import { useTranslation } from '../../../i18n';

type Props = {
  toggleNameDialog: (val: boolean) => void;
  togglePasswordChangeDialog: (val: boolean) => void;
};

export const AccountCard: React.FC<Props> = ({ toggleNameDialog, togglePasswordChangeDialog }) => {
  const [email, setEmail] = useState('someemail@mail.com');
  const [emailChangeMode, setEmailChangeMode] = useState(false);
  const { t } = useTranslation();

  const toggleEmailChangeMode = (val: boolean) => setEmailChangeMode(val);

  return (
    <div className={styles.accountContentWrapper}>
      <div className={styles.accountCard}>
        <div className={styles.avatar}></div>
        <div className={styles.cardActions}>
          <div className={styles.name}>Name Surname</div>
          <div className={styles.requestChange} onClick={() => toggleNameDialog(true)}>
            {t('account:requestChange')}
          </div>

          <div className={styles.textField}>
            <Input
              disabled={!emailChangeMode}
              value={email}
              endAdornment={
                <InputAdornment position='end'>
                  <IconButton onClick={() => toggleEmailChangeMode(!emailChangeMode)}>
                    <EditIcon color='primary' />
                  </IconButton>
                </InputAdornment>
              }
            />

            {emailChangeMode && (
              <Button fullWidth color='primary'>
                {t('account:saveEmail')}
              </Button>
            )}
          </div>

          <div className={styles.changePassword} onClick={() => togglePasswordChangeDialog(true)}>
            <span>{t('account:changePassword')}</span>
            <LockIcon color='primary' />
          </div>
        </div>
      </div>
    </div>
  );
};
