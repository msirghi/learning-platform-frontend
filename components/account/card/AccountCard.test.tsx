import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { AccountCard } from './AccountCard';

describe('AccountCard component', () => {
  const props = {
    toggleNameDialog: jest.fn(),
    togglePasswordChangeDialog: jest.fn()
  };

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should match the snapshot', () => {
    const wrapper = shallow(<AccountCard {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should call prop method on password change click', () => {
    const wrapper = shallow(<AccountCard {...props} />);
    const pwdChange = wrapper.find('.changePassword');

    expect(pwdChange).toHaveLength(1);

    pwdChange.simulate('click');
    expect(props.togglePasswordChangeDialog).toBeCalledTimes(1);
  });

  it('should call prop method on request change click', () => {
    const wrapper = shallow(<AccountCard {...props} />);
    const requestChange = wrapper.find('.requestChange');

    expect(requestChange).toHaveLength(1);

    requestChange.simulate('click');
    expect(props.toggleNameDialog).toBeCalledTimes(1);
  });
});
