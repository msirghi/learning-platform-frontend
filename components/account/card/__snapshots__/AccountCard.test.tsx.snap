// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountCard component should match the snapshot 1`] = `
<div
  className="accountContentWrapper"
>
  <div
    className="accountCard"
  >
    <div
      className="avatar"
    />
    <div
      className="cardActions"
    >
      <div
        className="name"
      >
        Name Surname
      </div>
      <div
        className="requestChange"
        onClick={[Function]}
      >
        account:requestChange
      </div>
      <div
        className="textField"
      >
        <WithStyles(ForwardRef(Input))
          disabled={true}
          endAdornment={
            <WithStyles(ForwardRef(InputAdornment))
              position="end"
            >
              <WithStyles(ForwardRef(IconButton))
                onClick={[Function]}
              >
                <Memo
                  color="primary"
                />
              </WithStyles(ForwardRef(IconButton))>
            </WithStyles(ForwardRef(InputAdornment))>
          }
          value="someemail@mail.com"
        />
      </div>
      <div
        className="changePassword"
        onClick={[Function]}
      >
        <span>
          account:changePassword
        </span>
        <Memo([object Object])
          color="primary"
        />
      </div>
    </div>
  </div>
</div>
`;
